projects/helper/chain/radixdlt.js

```diff
@@ -37,12 +40,57 @@ async function queryAddresses({ addresses = [], }) {
   return items
 }
 
+async function queryLsus(addresses = []) {
+  // Receives a list of lsu addresses and returns the redemption value in XRD
+
+  let lsuRedemptionValues = {}
+  const chunks  = sliceIntoChunks(addresses, 20)
+  for (const chunk of chunks) {
+    let body = {
+      "addresses": chunk,
+      "aggregation_level": "Vault"
+    }
+    let data = await post(ENTITY_DETAILS_URL, body)
+
+    let validators = []
+    for (const lsuResource of data.items) {
+      let validator = lsuResource.metadata.items.filter(metadataItem => metadataItem.key === "validator")[0]
+      lsuRedemptionValues[lsuResource.address] = {
+        "totalSupplyOfStakeUnits": lsuResource.details.total_supply,
+        "validatorAddress": validator.value.typed.value
+      }
+
+      validators.push(validator.value.typed.value)
+    }
+
+    let validatorsDetailsBody = {
+      "addresses": validators,
+      "aggregation_level": "Vault"
+    }
+    let validatorsData = await post(ENTITY_DETAILS_URL, validatorsDetailsBody)
+    for (const validator of validatorsData.items) {
```

Comment:
If the validator doesn't exist, do we handle that? I assume `validator` is null in that case?

Reply:
if the validator does not exist, we would get a 400 error from the gateway, which is handled down the road